core/src/main/java/io/questdb/cairo/vm/MemoryCARWSpillable.java

```diff
@@ -0,0 +1,283 @@
+package io.questdb.cairo.vm;
+
+import java.util.UUID;
+
+import org.jetbrains.annotations.NotNull;
+
+import io.questdb.cairo.CairoConfiguration;
+import io.questdb.cairo.vm.api.MemoryCARW;
+import io.questdb.log.Log;
+import io.questdb.log.LogFactory;
+import io.questdb.std.BinarySequence;
+import io.questdb.std.FilesFacade;
+import io.questdb.std.FilesFacadeImpl;
+import io.questdb.std.Long256;
+import io.questdb.std.MemoryTag;
+import io.questdb.std.Mutable;
+import io.questdb.std.Numbers;
+import io.questdb.std.Vect;
+import io.questdb.std.str.Path;
+
+public class MemoryCARWSpillable implements MemoryCARW, Mutable {
+    private static final Log LOG = LogFactory.getLog(MemoryCARWSpillable.class);
+    // for non-mmap
+    private final int maxPages;
+    // flag
+    private long diskSpillThreshold;
+    private long extendSegmentMsb;
+    private FilesFacade ff = new FilesFacadeImpl();
+    private boolean isSpilled;
+    private MemoryCARW mem;
+    private long minMappedMemorySize = -1;
+    private int nativeMemoryTag;
+    private int opts;
+    // general
+    private long pageSize;
+    private int spillMemoryTag;
+    // for mmap
+    private Path spillPath;
+    private String tmpFile;
+
+    // TODO: For now assume the same pageSize
+    public MemoryCARWSpillable(long pageSize, int maxPages, long diskSpillThreshold, CairoConfiguration cairoConf){
```

Comment:
No need to have so many constructors. It's enough to keep only the one needed by the `FastMap`. The same applies to all methods here - they can be `FastMap`-specific and there is no need to keep them if they're unused. In fact, this call doesn't even have to implement `MemoryCARW` especially considering that certain methods lose their meaning due to different underlying memories, e.g. `getExtendSegmentSize()`, `getPageCount()` or `getPageSize()`.

Reply:
I see..
Tbh, I found several inconsistencies inside MemoryCMARWImpl and MemoryCARWImpl where it might refer to different things (but most of the methods there such as `getExtendSegmentSize()` are derived from shared interface, so I believe they should convey the same meaning)

Reply:
Btw, how should I set the pageSize for each memory type (filebased and plain mem)?
Is it configurable?
if so, what should be sensible default? 4kb?

Reply:
For the "normal" (non-file-baked) memory, the page size should be the same as `FastMap`'s initial size. For the mmapped memory, page size doesn't matter - the extend segment size does. I'd make both of these configurable, so that `FastMap` passes them when initializing the heap object. Also, the extend segment size should be configurable through the `CairoConfiguration`. The default value may be something like 16MB since we know that disk spill has to deal with large files and frequent file extensions will be costly with 4KB.